confluent_kafka/avro/serializer/message_serializer.py

```diff
@@ -59,6 +59,30 @@ def __exit__(self, *args):
         return False
 
 
+class HasSchemaMixin:
+    """
+    Wraps a decoded Avro message dictionary to make able to add schema
+    """
+    def schema(self):
+        return self._schema
+
+
+def _wrap(value, schema):
+    if hasattr(schema, 'namespace'):
+        name = "{namespace}.{name}".format(namespace=schema.namespace,
+                                           name=schema.name)
+    elif hasattr(schema, 'name'):
+        name = schema.name
+    else:
+        name = schema.type
+
+    klass = type(str(name), (value.__class__, HasSchemaMixin), {})
```

Comment:
* isn't `name` already a string?
 * klass is swedish, use cls or similar.

Reply:
If we still support 2.7, is not `name` a formally unicode object?